test/integ_tests/test_create_quantum_job.py

```diff
@@ -97,27 +95,24 @@ def test_failed_quantum_job(aws_session, capsys):
     )
 
 
-def test_completed_quantum_job(aws_session, capsys):
+def test_completed_quantum_job(aws_session, capsys, completed_quantum_job):
     """Asserts the hybrid job is completed with the output, checkpoints, quantum tasks and
     script folder created in S3 for respective hybrid job. Validate the results are
     downloaded and results are what we expect. Also, assert that logs contains all the
     necessary steps for setup and running the hybrid job and is displayed to the user.
     """
 
-    job = AwsQuantumJob.create(
-        "arn:aws:braket:::device/quantum-simulator/amazon/sv1",
-        source_module="test/integ_tests/job_test_script.py",
-        entry_point="job_test_script:start_here",
-        wait_until_complete=True,
-        aws_session=aws_session,
-        hyperparameters={"test_case": "completed"},
-    )
-
+    job = completed_quantum_job
+    job_name = job.name
     pattern = f"^arn:aws:braket:{aws_session.region}:\\d{{12}}:job/[a-z0-9-]+$"
     assert re.match(pattern=pattern, string=job.arn)
 
     # check job is in completed state.
-    assert job.state() == "COMPLETED"
+    while True:
+        time.sleep(5)
+        if job.state() in AwsQuantumJob.TERMINAL_STATES:
+            break
+    assert job.state(use_cached_value=True) == "COMPLETED"
```

Comment:
Any reason to do this loop instead of calling `job.result()`?

Reply:
Fixed